overlord/configstate/configcore/certs.go

```diff
@@ -0,0 +1,136 @@
+	// This handles the "snap revert core" case:
+	// We need to go over each pem cert on disk and check if there is
+	// a matching config entry - if not->delete the cert
+	//
+	// XXX: remove this code once we have a general way to handle
+	//      "snap revert" and config updates
+	//
+	// TODO: add ways to detect cleanly if tr is a patch, skip the sync code if it is
+	storeCerts, err := filepath.Glob(filepath.Join(dirs.SnapdStoreSSLCertsDir, "*.pem"))
+	if err != nil {
+		return fmt.Errorf("cannot get exiting store certs: %v", err)
+	}
+	for _, storeCertPath := range storeCerts {
+		optionName := strings.TrimSuffix(filepath.Base(storeCertPath), ".pem")
+		v, err := coreCfg(tr, "store-certs."+optionName)
+		if err != nil {
+			return err
+		}
+		if v == "" {
+			if err := os.Remove(storeCertPath); err != nil {
+				return err
+			}
+		}
+	}
+
+	// add/remove regular (non revert) changes
+	for _, name := range tr.Changes() {
+		if !strings.HasPrefix(name, "core.store-certs.") {
+			continue
+		}
+
+		nameWithoutSnap := strings.SplitN(name, ".", 2)[1]
+		cert, err := coreCfg(tr, nameWithoutSnap)
+		if err != nil {
+			return fmt.Errorf("internal error: cannot get data for %s: %v", nameWithoutSnap, err)
+		}
+		optionName := strings.SplitN(name, ".", 3)[2]
+		certPath := filepath.Join(dirs.SnapdStoreSSLCertsDir, optionName+".pem")
+		switch cert {
+		case "":
+			// remove
+			if err := os.Remove(certPath); err != nil && !os.IsNotExist(err) {
+				return fmt.Errorf("cannot remove cert: %v", err)
+			}
+		default:
+			if err := os.MkdirAll(dirs.SnapdStoreSSLCertsDir, 0755); err != nil {
+				return fmt.Errorf("cannot create store ssl certs dir: %v", err)
+			}
+			if err := ioutil.WriteFile(certPath, []byte(cert), 0644); err != nil {
+				return fmt.Errorf("cannot write extra cert: %v", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func validateCertSettings(tr config.Conf) error {
+	for _, name := range tr.Changes() {
+		if !strings.HasPrefix(name, "core.store-certs.") {
+			continue
+		}
+
+		nameWithoutSnap := strings.SplitN(name, ".", 2)[1]
+		cert, err := coreCfg(tr, nameWithoutSnap)
+		if err != nil {
+			return fmt.Errorf("internal error: cannot get data for %s: %v", nameWithoutSnap, err)
+		}
+		if cert != "" {
+			optionName := strings.SplitN(name, ".", 3)[2]
+			if !validCertName(optionName) {
+				return fmt.Errorf("cannot set %q: name must only contain word characters or a dash", optionName)
+			}
+			// copied validation code from
+			// go:crypto/x509/cert_pool.go:AppendCertsFromPEM()
+			pemCerts := []byte(cert)
+			ok := false
+			for len(pemCerts) > 0 {
+				var block *pem.Block
+				block, pemCerts = pem.Decode(pemCerts)
+				if block == nil {
+					break
+				}
+				if block.Type != "CERTIFICATE" || len(block.Headers) != 0 {
+					continue
+				}
+				_, err := x509.ParseCertificate(block.Bytes)
+				if err != nil {
+					return fmt.Errorf("cannot parse pem certificate %q", optionName)
```

Comment:
maybe we should have a test that triggers this one

Reply:
Will look into this, *maybe* in a followup but I will see what I can do.

Reply:
Test added